domain-services/fiscal-event-post-processor/src/test/java/org/egov/service/FiscalEventFlattenServiceTest.java

```diff
@@ -0,0 +1,84 @@
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@SpringBootTest
+class FiscalEventFlattenServiceTest {
+
+    @InjectMocks
+    private FiscalEventFlattenService fiscalEventFlattenService;
+
+    @Test
+    void testGetFlattenDataWithEmptyUnbundledFiscalEvent() {
+        assertTrue(this.fiscalEventFlattenService.getFlattenData(new ArrayList<FiscalEventLineItemUnbundled>()).isEmpty());
+    }
+
+    @Test
+    void testGetFlattenDataWithDefaultUnbundledFiscalEvent() {
+        ArrayList<FiscalEventLineItemUnbundled> fiscalEventLineItemUnbundledList = new ArrayList<FiscalEventLineItemUnbundled>();
+        fiscalEventLineItemUnbundledList.add(new FiscalEventLineItemUnbundled());
+        List<String> actualFlattenData = this.fiscalEventFlattenService.getFlattenData(fiscalEventLineItemUnbundledList);
+        assertEquals(1, actualFlattenData.size());
+        assertEquals("{}", actualFlattenData.get(0));
+    }
+
+    @Test
+    void testGetFlattenDataWithNullUnbundledFiscalEvent() {
+        ArrayList<FiscalEventLineItemUnbundled> fiscalEventLineItemUnbundledList = new ArrayList<FiscalEventLineItemUnbundled>();
+        fiscalEventLineItemUnbundledList.add(null);
+        List<String> actualFlattenData = this.fiscalEventFlattenService.getFlattenData(fiscalEventLineItemUnbundledList);
+        assertEquals(1, actualFlattenData.size());
+        assertEquals("null", actualFlattenData.get(0));
+    }
+
+    @Test
+    void testGetFlattenData() {
+        ArrayList<FiscalEventLineItemUnbundled> fiscalEventLineItemUnbundledList = new ArrayList<FiscalEventLineItemUnbundled>();
+        Government government = new Government("42", "Name");
+
+        Department department = new Department("42", "Code", "Name");
+
+        DepartmentEntity departmentEntity = new DepartmentEntity();
+        Expenditure expenditure = new Expenditure("42", "Code", "Name", Expenditure.TypeEnum.SCHEME);
+
+        Project project = new Project("42", "Code", "Name");
+
+        BigDecimal amount = BigDecimal.valueOf(1L);
+        ChartOfAccount coa = new ChartOfAccount("42", "Coa Code", "Major Head", "Major Head Name", "Major Head Type",
+                "Sub Major Head", "Sub Major Head Name", "Minor Head", "Minor Head Name", "Sub Head", "Sub Head Name",
+                "Group Head", "Group Head Name", "Object Head", "Object Head Name");
+
+        fiscalEventLineItemUnbundledList
+                .add(new FiscalEventLineItemUnbundled("1.0.2", "42", "42", "42", government, department, departmentEntity,
+                        expenditure, project, "Event Type", 1L, 1L, "42", "42", "42", amount, coa, 1L, 1L, new AuditDetails()));
+        List<String> actualFlattenData = this.fiscalEventFlattenService.getFlattenData(fiscalEventLineItemUnbundledList);
+        assertEquals(1, actualFlattenData.size());
+        assertEquals(
+                "{\"version\":\"1.0.2\",\"id\":\"42\",\"eventId\":\"42\",\"tenantId\":\"42\",\"government.id\":\"42\",\"government.name\":"
```

Comment:
Can we move this to a .json file in the resources directory?

Reply:
as discussed ,resolved.